### DISCUSSION TITLE
unexpectedly high CPU usage with H264Encoder on RPi Zero

### DISCUSSION THREAD

**Discussion by jimfum** [raspberrypi/picamera2#601] [General]
I've posted a similar Q to Raspberry Pi forum (https://forums.raspberrypi.com/viewtopic.php?p=2086996#p2086996) but could belong here as well, not sure.

I just installed the latest Bullseye LITE on RPI Zero v1.3 and narrowed down my testing to the below simple code that just doesn't do anything apart from starting h264 capture but without storing the data anywhere. It spins up 6 processes for this (as per `htop`) and the CPU utilization goes to 42%-70%, with a default framerate (25 or 30?) to 85%.

There must be something wrong. Buster and `picamera` with such code but extended to actually stream the data to the network used something like 18% CPU.

I must have overlooked something.

```python
from picamera2 import Picamera2
from picamera2.encoders import H264Encoder
from picamera2.outputs import FileOutput
import io

class StreamingOutput(io.BufferedIOBase):
    def write(self, buf):
        pass

try:
    picam2 = Picamera2()
    video_config = picam2.create_video_configuration(main={"size": (1280, 720)}, controls={"FrameRate":15})
    picam2.configure(video_config)
    output = StreamingOutput()
    picam2.start_recording(H264Encoder(bitrate=1000000), FileOutput(output))
    input()
finally:
    picam2.stop_recording()
```

**Comment by davidplowman** [comment]
Hi, the general direction of camera stacks is to run more open source code on the Arm processors which unfortunately makes this hard to avoid. It may help somewhat to set the main stream format to "YUV420" as this will save both memory and memory bandwidth, otherwise you may need to consider continuing with the legacy camera stack.

**Comment by jimfum** [comment]
Thanks David, that's understood. "YUV420" doesn't seem to have a noticeable effect.
